lang/java/reef-common/src/main/java/org/apache/reef/runtime/common/launch/JavaLaunchCommandBuilder.java

```diff
@@ -37,19 +37,22 @@
  * Build the launch command for Java REEF processes.
  */
 public final class JavaLaunchCommandBuilder implements LaunchCommandBuilder {
+
   private static final Logger LOG = Logger.getLogger(JavaLaunchCommandBuilder.class.getName());
 
-  private static final String DEFAULT_JAVA_PATH = System.getenv("JAVA_HOME") + "/bin/" + "java";
+  private static final String DEFAULT_JAVA_PATH = "java"; // Assume java binary is in the PATH
```

Comment:
I've just run our HelloREEF application on CISL YARN cluster with `{{JAVA_HOME}}/bin/java` and then with `%JAVA_HOME%/bin/java` - and in **both** cases it works! I bet Spark just uses one of those notations - will give you a definite answer in a minute